### PR TITLE
disk-format:  New checkpoint for snapshot commitment

### PR DESCRIPTION
**Pull request by hellohellenmao**
id:  1719107

disk-format: New checkpoint - check the snapshot whether emptied after commitment

Signed-off-by: Tingting Mao <timao@redhat.com>

### CODE REVIEW

**Comment by zhenyzha** [comment]
git am compile failed
Manual compile failed too.

**Comment by zhenyzha** [comment]
http://10.0.136.47/zhenyzha/0618qemu_disk_img
From 49 tests executed, 48 passed and 0 warned - success rate of 97.96% 

14:25:00 INFO | Command 'dd if=/dev/urandom of=/var/tmp/test.img bs=1M count=1024' finished with 0 after 3.007737159729004s
14:25:00 DEBUG| [stderr] 1024+0 records out
14:25:00 DEBUG| [stderr] 1073741824 bytes (1.1 GB, 1.0 GiB) copied, 3.00603 s, 357 MB/s
14:25:56 INFO | Verify the snapshot file whether emptied after committing
14:25:56 INFO | The snapshot file was emptied!
-----------------------------------------------------------------------------------------------------------------
http://10.0.136.47/zhenyzha/0618commit_snapshot_to_raw_backing
From 7 tests executed, 7 passed and 0 warned - success rate of 100.00% 

20:35:49 INFO | Command 'dd if=/dev/urandom of=/var/tmp/testfile bs=1M count=1024' finished with 0 after 5.156603097915649s
20:35:49 DEBUG| [stderr] 1024+0 records out
20:35:49 DEBUG| [stderr] 1073741824 bytes (1.1 GB, 1.0 GiB) copied, 5.15534 s, 208 MB/s
20:36:51 INFO | Verify the snapshot file whether emptied after committing
20:36:51 INFO | The snapshot file was emptied!

Acked-by: Zhenyu Zhang <zhenyzha@redhat.com>

File: qemu/tests/commit_snapshot_to_raw_backing.py [hunk 1]
```diff
@@ -17,7 +17,7 @@ def run(test, params, env):
     2. boot the guest from the snapshot
     3. create a file in the snapshot disk, calculate its md5sum
     4. shut the guest down
-    5. commit the snapshot to the raw backing image with various cache mode
+    5. commit the snapshot to the raw backing image with various cache mode,and check whether the snapshot file emptied
```
Comment (hereischen) [inline qemu/tests/commit_snapshot_to_raw_backing.py]: please break to a new line, this line is too long

**Comment by hellohellenmao** [comment]
In x86:
    For raw:
    (3/3) Host_RHEL.m8.u1.raw.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.commit_snapshot_to_raw_backing.compat_level.compat_default: PASS (204.42 s)

    For qcow2:
    (3/3) Host_RHEL.m8.u1.qcow2.virtio_scsi.up.virtio_net.Guest.RHEL.7.6.x86_64.io-github-autotest-qemu.qemu_disk_img.commit: PASS (192.28 s)

**Comment by hereischen** [comment]
Acked-by: Haotong Chen hachen@redhat.com

File: qemu/tests/commit_snapshot_to_raw_backing.py [hunk 2]
```diff
@@ -80,7 +81,21 @@ def _verify_qemu_img_info(output, b_fmt, b_name):
...
+    org_size = json.loads(sn_img.info(output="json"))["actual-size"]
     sn_img.commit(cache_mode=cache_mode)
+    remain_size = json.loads(sn_img.info(output="json"))["actual-size"]
+
+    """Verify the snapshot file whether emptied after committing"""
+    logging.info("Verify the snapshot file whether emptied after committing")
+    commit_size = org_size - remain_size
+    dd_bs = int(params["dd_bs_val"])
+    dd_count = int(params["dd_bs_count"])
+    dd_size = eval('dd_bs * dd_count * 1024 * 1024')
```
Comment (YongxueHong) [inline qemu/tests/commit_snapshot_to_raw_backing.py]: For line 92~94, you could do like this:
`dd_size = eval('params["dd_bs_val"] * params["dd_bs_count"] * 1024 ** 2')`

File: qemu/tests/qemu_disk_img_commit.py [hunk 3]
```diff
@@ -51,9 +61,26 @@ def run(test, params, env):
...
     error_context.context("sync host data after commit", logging.info)
     process.system("sync")
+    remain_size = json.loads(sn_img.info(output="json"))["actual-size"]
+
+    """Verify the snapshot file whether emptied after committing"""
+    logging.info("Verify the snapshot file whether emptied after committing")
+    commit_size = org_size - remain_size
+    dd_bs = int(params["dd_bs_val"])
+    dd_count = int(params["dd_bs_count"])
+    dd_size = eval('dd_bs * dd_count * 1024 * 1024')
```
Comment (YongxueHong) [inline qemu/tests/qemu_disk_img_commit.py]: Also, like the above commit.

**Pull request merged**